src/types.rs

```diff
@@ -8,15 +8,30 @@ use tracing::field::Visit;
 use tracing::span::{Attributes, Id};
```

Comment:
fwiw, personally, having a `mod types` to put all your types in seems very weird to me...I'd prefer to group types and functions into modules based on their purpose/function. could be a matter of taste.